daemon/build.go

```diff
@@ -75,15 +76,24 @@ func (rl *releaseableLayer) Release() error {
 	if rl.released {
 		return nil
 	}
+	if err := rl.releaseRWLayer(); err != nil {
+		return err
+	}
+	if err := rl.releaseROLayer(); err != nil {
```

Comment:
It seems weird that if this fails, we have released the RW layer but didn't remember this fact.